modules/concept-docs/pages/n1ql-query.adoc

```diff
@@ -6,12 +6,22 @@
 [abstract]
 Parallel data management for complex queries over many records, using a familiar SQL-like syntax.
 
-The N1QL Query Language provides a familiar, SQL-like experience for querying documents stored in Couchbase.
-You can xref:6.5@server:n1ql:n1ql-language-reference/index.adoc[read up on the language in our reference guide], 
-but you probably just want to xref:howtos:n1ql-queries-with-sdk.adoc[dive into a practical example].
 
-Below, we fill in some of the gaps between reference and rolling-up-your-sleeves practicality,
-with discussion of a few areas of the Query service where more background knowledge will help you to better program your application.
+include::6.5@sdk:shared:partial$n1ql-queries.adoc[tag=intro]
+
+
+// Prepared Statements for Query Optimization
+include::6.5@sdk:shared:partial$n1ql-queries.adoc[tag=prepared]
+
+Do not turn off the `adhoc` flag for _every_ query to Server 6.0 and earlier, since only a finite number of query plans (currently 5000) can be stored in the SDK.
```

Comment:
This sounds a little too prohibitive I think.. Maybe more along the lines of "If your queries are highly dynamic, we recommend using parameterized queries if possible (epecially when prepared statements are not used). Parameterized queries are more cache efficient and will allow for better performance. The maximum client side query cache size is 5000 entries.